dandi/files.py

```diff
@@ -768,7 +774,8 @@ def iter_upload(
         :param int jobs: Number of threads to use for uploading; defaults to 5
         :param RemoteAsset replacing:
             If set, replace the given asset, which must have the same path as
-            the new asset
+            the new asset; if the old asset is a Zarr, the Zarr will be updated
+            & reused for the new asset
```

Comment:
ha. Originally I thought we would just keep modifying the zarr, and thus leaving the asset the same, not a new one upon each attempt to finish up upload or update it.  API allows for that but I think having a new asset to replace an old one if we are to change zarr, makes sense.

WDYT @satra -- may be I am missing some use case/aspect where we do want to keep the asset to be the same?

Reply:
@jwodder - that's essentially what i wanted. so seems like nothing needs to change. then in that case if the upload crashes, does the CLI check that an existing asset is there, retrieves the zarr id and uploads to that location?

regarding @yarikoptic comment, i don't think assets need to remain the same. however, my point about when a new asset is minted and who mints it still needs to be resolved with respect to updates that the server does (checksum, contentsize, timestamps).

Reply:
@satra 

> then in that case if the upload crashes, does the CLI check that an existing asset is there, retrieves the zarr id and uploads to that location?

Yes.

Reply:
> when a new asset is minted and who mints it still needs to be resolved with respect to updates that the server does (checksum, contentsize, timestamps).

indeed a good question, and something worth asking @dchiquito : if we e.g. mint an asset for an empty zarr, then upload into that zarr, would that asset get its record and metadata updated with new size/checksum upon zarr's upload `/complete`? (feels unlikely, but possible, didn't check)